bcipy/helpers/task.py

```diff
@@ -414,32 +411,29 @@ def __call__(self,
         timing_info = tmp_timing
 
         n_inquiry = len(timing_info) // trials_per_inquiry
+        trial_duration_samples = int(trial_length * fs)
 
         # triggers in seconds are mapped to triggers in number of samples.
         triggers = list(map(lambda x: int((x + offset) * fs), timing_info))
 
-        # shape (Channels, Inquiries, Samples)
-        reshaped_data = np.zeros((len(eeg_data), n_inquiry, num_samples))
-
         # Label for every inquiry
-        labels = np.zeros(n_inquiry)
-        for inquiry_idx, chunk in enumerate(grouper(zip(trial_labels, triggers), trials_per_inquiry)):
+        labels = np.zeros(n_inquiry, dtype=np.long)
+        reshaped_data = []
+        for inquiry_idx, trials_within_inquiry in enumerate(grouper(zip(trial_labels, triggers), trials_per_inquiry)):
             # label is the index of the "target", or else the length of the inquiry
             inquiry_label = trials_per_inquiry
-            first_trigger = None
-            for trial_idx, (trial_label, trigger) in enumerate(chunk):
-                if first_trigger is None:
-                    first_trigger = trigger
-
+            for trial_idx, (trial_label, trigger) in enumerate(trials_within_inquiry):
```

Comment:
I feel like Python is missing an abstraction that would help here. It has the `index` method for lists, but this method doesn't allow for a `key` parameter to choose how to compare items (like the `sorted` function). Something like:

    def index_of(lst, element, key=None):
        for i, item in enumerate(lst): 
            if (key and key(item) == element) or (not key and item == element): 
                return i 
        return None

Then the code here could be:
 
    inquiry_label = index_of(trials_within_inquiry, target_label, key=operator.itemgetter(0)) or trials_per_query

which almost matches the comment. Here's an alternate implementation:

    def index_of(lst, element, key=None):
        predicate = (lambda item: key(item) == element) if key else (lambda item: item == element)
        for i, item in enumerate(lst): 
            if predicate(item): 
                return i 
        return None

Reply:
Interesting idea! I haven't written C code since compilers class, but this sounds like it would be a great contribution to python source btw...